newIDE/app/src/QuickCustomization/QuickCustomizationPropertiesVisibilityDialog.js

```diff
@@ -0,0 +1,116 @@
+// @flow
+import { t, Trans } from '@lingui/macro';
+
+import * as React from 'react';
+import Dialog from '../UI/Dialog';
+import FlatButton from '../UI/FlatButton';
+import { mapFor } from '../Utils/MapFor';
+import Text from '../UI/Text';
+import SelectField from '../UI/SelectField';
+import SelectOption from '../UI/SelectOption';
+import { LineStackLayout } from '../UI/Layout';
+import useForceUpdate from '../Utils/UseForceUpdate';
+
+const gd: libGDevelop = global.gd;
+
+const getVisibilityForProperty = (
+  propertyName: string,
+  propertiesQuickCustomizationVisibilities: gdQuickCustomizationVisibilitiesContainer
+) => {
+  return propertiesQuickCustomizationVisibilities.get(propertyName);
+};
+
+type Props = {|
+  propertyNames: string[],
+  open: boolean,
+  onClose: () => void,
+  propertiesQuickCustomizationVisibilities: gdQuickCustomizationVisibilitiesContainer,
+|};
+
+export default function QuickCustomizationPropertiesVisibilityDialog({
+  open,
+  onClose,
+  propertyNames,
+  propertiesQuickCustomizationVisibilities,
+}: Props) {
+  console.log(propertyNames);
+  const forceUpdate = useForceUpdate();
+
+  return (
+    <Dialog
+      title={<Trans>Quick Customization: Behavior properties</Trans>}
+      secondaryActions={[
+        <FlatButton
+          key="close"
+          label={<Trans>Close</Trans>}
+          primary={false}
+          onClick={onClose}
+        />,
+      ]}
+      open
+      onRequestClose={onClose}
+      flexColumnBody
+      fullHeight
+    >
+      {mapFor(0, propertyNames.length, i => {
+        const propertyName = propertyNames[i];
+        console.log('propertyName:', propertyName);
+        const value = getVisibilityForProperty(
+          propertyName,
+          propertiesQuickCustomizationVisibilities
+        );
+
+        return (
+          <LineStackLayout
+            key={propertyName}
+            justifyContent="flex-start"
+            alignItems="center"
+          >
+            <Text>{propertyName}</Text>
+            <SelectField
+              margin="none"
+              value={value}
+              onChange={(e, i, newValue: string) => {
+                const newQuickCustomizationVisibility = parseInt(newValue, 10);
+                if (
+                  [
+                    gd.QuickCustomization.Visible,
+                    gd.QuickCustomization.Hidden,
+                    gd.QuickCustomization.Default,
+                  ].includes(newQuickCustomizationVisibility)
+                ) {
+                  propertiesQuickCustomizationVisibilities.set(
+                    propertyName,
+                    // $FlowIgnore: We checked that newQuickCustomizationVisibility is a valid visibility
+                    newQuickCustomizationVisibility
+                  );
+                  forceUpdate();
+                  return;
+                }
+
+                propertiesQuickCustomizationVisibilities.set(
```

Comment:
I went for a set prop by prop, maybe we'd rather:
store the values in state & have a "Save" button in the Dialog that clears the Map & sets again all the props.
This would avoid possible values in the future not being removed if they change name

Reply:
I think that's fine for now, but you're right that a full clear+set could be useful later.